pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -23,20 +23,181 @@ import (
 	"path"
 	"strings"
 
-	"k8s.io/kubernetes/pkg/volume"
+	k8s_volume "k8s.io/kubernetes/pkg/volume"
 
 	"github.com/gophercloud/gophercloud"
 	"github.com/gophercloud/gophercloud/openstack"
-	"github.com/gophercloud/gophercloud/openstack/blockstorage/v1/volumes"
+	volumes_v1 "github.com/gophercloud/gophercloud/openstack/blockstorage/v1/volumes"
+	volumes_v2 "github.com/gophercloud/gophercloud/openstack/blockstorage/v2/volumes"
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/volumeattach"
 	"github.com/gophercloud/gophercloud/pagination"
 
 	"github.com/golang/glog"
 )
 
+type IVolumes interface {
+	Create(opts VolumeCreateOpts) (string, error)
+	getVolume(diskName string) (Volume, error)
+	Delete(volumeName string) error
+}
+
+// Volumes implementation for v1
+type VolumesV1 struct {
+	blockstorage *gophercloud.ServiceClient
+	opts         BlockStorageOpts
+}
+
+// Volumes implementation for v2
+type VolumesV2 struct {
+	blockstorage *gophercloud.ServiceClient
+	opts         BlockStorageOpts
+}
+
+type Volume struct {
+	// ID of the instance, to which this volume is attached. "" if not attached
+	AttachedServerId string
+	// Device file path
+	AttachedDevice string
+	// Unique identifier for the volume.
+	ID string
+	// Human-readable display name for the volume.
+	Name string
+	// Current status of the volume.
+	Status string
+}
+
+type VolumeCreateOpts struct {
+	Size         int
+	Availability string
+	Name         string
+	VolumeType   string
+	Metadata     map[string]string
+}
+
+func (volumes *VolumesV1) Create(opts VolumeCreateOpts) (string, error) {
+
+	create_opts := volumes_v1.CreateOpts{
+		Name:         opts.Name,
+		Size:         opts.Size,
+		VolumeType:   opts.VolumeType,
+		Availability: opts.Availability,
+		Metadata:     opts.Metadata,
+	}
+
+	vol, err := volumes_v1.Create(volumes.blockstorage, create_opts).Extract()
+	return vol.ID, err
+}
+
+func (volumes *VolumesV2) Create(opts VolumeCreateOpts) (string, error) {
```

Comment:
It looks to me, except in `getVolume` where `ServerID` and `Device` are new in V2, `CreateDisk`, `DeleteDisk` and rest of `getVolume` are identical in both V1 and V2. 

In such case, why not just pass version as a new parameter into these functions and call the right package based on version, rather than copying code?